### PR TITLE
Auto test modules

### PR DESCRIPTION
**Pull request by zhichengzhang1995**
Auto test modules

### CODE REVIEW

File: test/os-queens-environment.json [hunk 1]
```diff
@@ -98,7 +98,7 @@
 		},
 		{
 			"key": "ADC_IMAGE",
-			"value": "13.1.1",
+			"value": "BIGIP-13.1.1",
```
Comment (zhaoqin-github) [inline test/os-queens-environment.json]: Can you please modify image name in another PR? This is not a part of your work, and we need this change right now.

File: test/Auto/shared-queens.rc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# The catalog contains the endpoints for all services the user/tenant has
+# access to - such as Compute, Image Service, Identity, Object Storage, Block
+# Storage, and Networking (code-named nova, glance, keystone, swift,
+# cinder, and neutron).
+#
+# *NOTE*: Using the 3 *Identity API* does not necessarily mean any other
+# OpenStack API is version 3. For example, your cloud provider may implement
+# Image API v1.1, Block Storage API v2, and Compute API v2.0. OS_AUTH_URL is
+# only for the Identity API served through keystone.
+export OS_AUTH_URL=http://10.250.13.69:5000/v3
```
Comment (zhaoqin-github) [inline test/Auto/shared-queens.rc]: Those environment settings can be set in Travis varaibles.

File: test/Auto/build.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+echo "Clone repo start"
+sudo git clone -b Auto https://github.com/zhichengzhang1995/f5-adcaas-openstack.git
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: Please remove sudo.

File: test/Auto/build.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+sudo mv ../TrustedProxy-1.0.1-0002.noarch.rpm ./dependencies/ASGExtensions/
+
+cd scripts
+echo "Start test"
+sudo ./start_all.sh
+
+echo "Test app"
+cd ../test
+# sudo npm install -g newman newman-reporter-html
+sudo ./test_all.sh
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: Your pr will not need to guarantee test suite must success. I suggest you can skip the test suite. Just provision the environment, start the apps, and make sure the apps are running.

Comment (zhaoqin-github) [reply test/Auto/build.sh]: We can add new steps to run test suite in the future.

File: test/Auto/build.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+sudo apt install docker.io -y
+
+echo "Typescript install start"
+sudo npm install -g typescript -y
+
+echo "Docker compose install start"
+# sudo curl -L "https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+# sudo chmod +x /usr/local/bin/docker-compose
+
+sudo npm install --save-dev eslint-config-prettier@^5.0.0
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: Why we need this?

File: test/Auto/build.sh [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+# sudo npm install --save-dev eslint@^1.1.2
+
+sudo systemctl start docker
+
+echo "App build and install start"
+cd f5-adcaas-openstack/app/waf
+
+echo "Npm install start"
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: do not need to run "npm install" and "npm build". Those will be done by docker build.

File: test/Auto/build.sh [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+sudo npm install
+
+echo "Npm build start"
+sudo npm run build
+
+echo "Portal install start"
+sudo docker build . -t f5devcentral/f5-openstack-services:portal-latest
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: Remove sudo.  You may add ubuntu user to docker group, if it can not run docker command.

File: test/Auto/build.sh [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+echo "Latest install start"
+sudo docker build . -t f5devcentral/f5-openstack-services:adcaas-latest
+
+echo "Start app"
+cd ../../
+
+# install rpm
+sudo mkdir -p dependencies/ASGExtensions/
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: remove sudo

File: .travis.yml [hunk 9]
```diff
@@ -25,13 +25,17 @@ jobs:
...
       before_script:
         - pushd test/vpn
         - ./vpn_install.sh
         - ./vpn_connect.sh
         - popd
       script:
-        - echo "No test"
+        - pushd test/Auto
+        - ./auto_test.sh
+        - ./auto_clean.sh
```
[the remark this answers is not in the document]
Comment (zhichengzhang1995) [reply .travis.yml]: fixed.

File: test/Auto/auto_clean.sh [hunk 10]
```diff
@@ -0,0 +1,2 @@
+source ./shared-queens.rc
+sudo openstack server delete test_instance
```
[the remark this answers is not in the document]
Comment (zhichengzhang1995) [reply test/Auto/auto_clean.sh]: fixed.

File: test/Auto/build.sh (hunk 5 above)
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/Auto/build.sh [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+echo "Npm install start"
+sudo apt-get install npm -y
+
+echo "Docker install start"
+sudo apt-get install apt-transport-https ca-certificates curl gnupg-agent     software-properties-common -y
+sudo apt install docker.io -y
+
+echo "Typescript install start"
+sudo npm install -g typescript -y
```
[the remark this answers is not in the document]
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/Auto/auto_test.sh [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+sudo pip install --upgrade pip
+sudo pip install --ignore-installed python-openstackclient
+
+openstack server create --flavor m1.large --image AutotestS2 --nic net-id=e12b30b7-d0d1-47f2-9be6-346913749ebc --key-name zz test_instance --config-drive True
+# create server
+# AutotestSS - Basic
+# AutotestS1 - With rpm
+# AutotestS2 - With all build
+
+sudo chmod 600 zz.pem
```
[the remark this answers is not in the document]
Comment (zhichengzhang1995) [reply test/Auto/auto_test.sh]: fixed.

File: test/Auto/build.sh (hunk 4 above)
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/os-queens-environment.json (hunk 1 above)
Comment (zhichengzhang1995) [reply test/os-queens-environment.json]: fixed

File: test/Auto/shared-queens.rc (hunk 2 above)
Comment (zhichengzhang1995) [reply test/Auto/shared-queens.rc]: fixed.

File: test/Auto/auto_test.sh [hunk 13]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+source ./shared-queens.rc
+sudo apt install python-dev python-pip
+sudo pip install --upgrade pip
+sudo pip install --ignore-installed python-openstackclient
+
+openstack server create --flavor m1.large --image $TEST_IMAGE --nic net-id=e12b30b7-d0d1-47f2-9be6-346913749ebc --key-name zz test_instance_$name --config-drive True
```
Comment (zhaoqin-github) [inline test/Auto/auto_test.sh]: Network id value need to be set in Travis

File: test/Auto/auto_test.sh (hunk 12 above)
[the remark this answers is not in the document]
Comment (zhaoqin-github) [reply test/Auto/auto_test.sh]: remove chmod

File: test/Auto/build.sh [hunk 14]
```diff
@@ -0,0 +1,66 @@
+echo "Clone repo start"
+sudo git clone https://github.com/F5Networks/f5-adcaas-openstack.git
+# sudo git clone -b Auto https://github.com/zhichengzhang1995/f5-adcaas-openstack.git
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: remove

File: test/Auto/build.sh [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+echo "Npm install start"
+sudo apt-get install npm -y
+
+echo "Docker install start"
+sudo apt-get install apt-transport-https ca-certificates curl gnupg-agent     software-properties-common -y
+sudo apt install docker.io -y
+
+echo "Typescript install start"
+# sudo npm install -g typescript -y
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: remove

File: test/Auto/build.sh [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+echo "Docker compose install start"
+# sudo curl -L "https://github.com/docker/compose/releases/download/1.24.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: install docker-compose, if it is not there

File: test/Auto/build.sh [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+# sudo chmod +x /usr/local/bin/docker-compose
+
+# sudo npm install --save-dev eslint-config-prettier@^5.0.0
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: remove

File: test/Auto/build.sh [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+sudo mv ../TrustedExtensions-1.0.1-0001.noarch.rpm ./dependencies/ASGExtensions/
+sudo mv ../TrustedProxy-1.0.1-0002.noarch.rpm ./dependencies/ASGExtensions/
+
+cd scripts
+echo "Start test"
+sudo ./start_all.sh
+
+echo "Test app"
+cd ../test
+# sudo npm install -g newman newman-reporter-html
```
Comment (zhaoqin-github) [inline test/Auto/build.sh]: Install newman, if it is not there. And I also suggest you to place all dependency installation in one place, so we will know them easily.

Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed

File: test/Auto/build.sh (hunk 17 above)
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/Auto/build.sh (hunk 15 above)
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/Auto/build.sh (hunk 14 above)
Comment (zhichengzhang1995) [reply test/Auto/build.sh]: fixed.

File: test/Auto/auto_test.sh (hunk 13 above)
Comment (zhichengzhang1995) [reply test/Auto/auto_test.sh]: fixed.

File: .travis.yml [hunk 19]
```diff
@@ -1,5 +1,3 @@
-language: node_js
```
Comment (zhaoqin-github) [inline .travis.yml]: what is the purpose to remove this?

File: .travis.yml [hunk 20]
```diff
@@ -73,4 +78,4 @@ jobs:
           branch: master
           repo: F5Networks/f5-adcaas-openstack
       after_deploy:
-         - aws cloudfront create-invalidation --distribution-id $AWS_DIST_ID --paths $AWS_DIR
+         - aws cloudfront create-invalidation --distribution-id $AWS_DIST_ID --paths $AWS_DIR
```
Comment (zhaoqin-github) [inline .travis.yml]: newline

File: .travis.yml [hunk 21]
```diff
@@ -25,13 +25,18 @@ jobs:
         script:
           - scripts/publish_api_ref.sh
     - stage: fvt
+      # if: TRAVIS_EVENT_TYPE = cron
```
Comment (zhaoqin-github) [inline .travis.yml]: Please enable this condition check in your next commit.

File: .travis.yml [hunk 22]
```diff
@@ -25,13 +25,18 @@ jobs:
         script:
           - scripts/publish_api_ref.sh
     - stage: fvt
+      if: TRAVIS_EVENT_TYPE = cron 
       before_script:
         - pushd test/vpn
         - ./vpn_install.sh
         - ./vpn_connect.sh
-        - popd
+        - popd
```
Comment (zhaoqin-github) [inline .travis.yml]: white space

**Pull request merged**